src/store/user/userData.api.ts

```diff
@@ -0,0 +1,221 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from 'firebase/auth';
+import { DocumentData } from 'firebase/firestore';
+
+import { AppDispatch, RootState } from '../store';
+
+import {
+  checkDocsInDB,
+  checkFieldInDocsInDB,
+  setDocsInDB,
+  signInWithPopupApi,
+  signOutApi,
+} from 'api/api';
+import { Fields, Collections } from 'api/types';
+import { handleError } from 'helpers/handleError';
+
+export interface IInitialState {
+  user: Nullable<User>;
+  token: Nullable<string>;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+  isNameAvailable: boolean;
+  error: Nullable<string | undefined>;
+}
+
+const initialState: IInitialState = {
+  user: null,
+  token: null,
+  isLoggedIn: false,
+  isLoading: false,
+  isNameAvailable: true,
+  error: null,
+};
+
+export const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(checkDocs.pending, state => {
+        state.isLoading = true;
+        state.error = null;
+      })
```

Comment:
try use [addMatcher](https://redux-toolkit.js.org/api/createreducer#builderaddmatcher) for similar in processing redux-thunk